runtime/common/src/claims.rs

```diff
@@ -147,18 +132,38 @@ impl sp_std::fmt::Debug for EcdsaSignature {
 	}
 }
 
-decl_event!(
-	pub enum Event<T> where
-		Balance = BalanceOf<T>,
-		AccountId = <T as frame_system::Config>::AccountId
-	{
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{pallet_prelude::*, traits::EnsureOrigin, weights::{Pays, DispatchClass}};
+	use frame_system::{pallet_prelude::*, ensure_signed, ensure_root, ensure_none};
+	use super::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	/// Configuration trait.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type VestingSchedule: VestingSchedule<Self::AccountId, Moment=Self::BlockNumber>;
+		#[pallet::constant]
+		type Prefix: Get<&'static [u8]>;
```

Comment:
Here introduced a metadata difference compared to FRAME v1 version. The constant type should be `&[u8]`, however there is no way to change the type in metadata(if I didn't missing any v2 new syntax). Need to wait for https://github.com/paritytech/substrate/issues/8827 being solved.

Reply:
does this metadata change really matter? my guess is probably no?

Reply:
The change will make `Prefix` type as `& 'static[u8]`. Not sure if polkadotjs handles this type, and if not, I think it would result a `createTypes` error(which would probably break apps). @jacogr could you please help to confirm?

Reply:
I just checked CI. Yes there's a `createType` error, and polkadotjs API can't be initialized because of it https://gitlab.parity.io/parity/polkadot/-/jobs/928510

> 2021-05-16 15:57:58        REGISTRY: Unable to resolve type &'static[u8], it will fail on construction
> 2021-05-16 15:57:58        API/INIT: Error: FATAL: Unable to initialize the API: createType(&'static[u8]):: Cannot construct unknown type &'static[u8]

Reply:
https://github.com/polkadot-js/api/pull/3545

In general, and not getting in the way here, this one I have mixed feelings about -

1. It needs to be done, since we want all converted
2. It is absolutely a nightmare to test this end-to-end to ensure nothing has been broken (so if something does break, this becomes a support horror)
3. It is not very active (most have claimed), but when used it just need to work flawlessly

So all-in-all, needs to be done, but not quite seeing the overall benefit.
3.